src/gei/Commands/GenerateArchiveCommand.cs

```diff
@@ -0,0 +1,96 @@
+        {
+            _log = log;
+            _sourceGithubApiFactory = sourceGithubApiFactory;
+            _environmentVariableProvider = environmentVariableProvider;
+
+            Description = "Invokes the GitHub Migration API's to generate a migration archive";
+            Description += Environment.NewLine;
+            Description += "Note: Expects GH_PAT and GH_SOURCE_PAT env variables to be set. GH_SOURCE_PAT is optional, if not set GH_PAT will be used instead. This authenticates to source GHES API";
+
+            var githubURL = new Option<string>("--github-url")
+            {
+                IsRequired = false
+            };
+            var githubSourceOrg = new Option<string>("--github-source-org")
+            {
+                IsRequired = true
+            };
+            var githubSourceRepo = new Option<string>("--github-source-repo")
+            {
+                IsRequired = true
+            };
+            var verbose = new Option("--verbose")
+            {
+                IsRequired = false
+            };
+
+            AddOption(githubURL);
+            AddOption(githubSourceOrg);
+            AddOption(githubSourceRepo);
+            AddOption(verbose);
+
+            Handler = CommandHandler.Create<string, string, string, bool>(Invoke);
+        }
+
+        public async Task Invoke(string githubURL, string githubSourceOrg, string githubSourceRepo, bool verbose = false)
+        {
+            _log.Verbose = verbose;
+
+            _log.LogInformation("Generating Migration Archive...");
+            _log.LogInformation($"GITHUB SOURCE ORG: {githubSourceOrg}");
+
+            if (string.IsNullOrWhiteSpace(githubURL))
+            {
+                _log.LogInformation("GithubURL not provided, defaulting to https://api.github.com");
+                githubURL = "https://api.github.com";
+            }
+
+            var repositories = new string[] { githubSourceRepo };
+
+            var githubApi = _sourceGithubApiFactory.CreateDefaultClient();
+            var migrationId = await githubApi.StartArchiveGeneration(githubURL, githubSourceOrg, repositories);
+
+            _log.LogInformation($"Archive generation started with id: {migrationId}");
+
+            var isFinished = false;
+            var timeOut = DateTime.Now.AddHours(10);
+
+            while (!isFinished && DateTime.Now < timeOut)
+            {
+                var archiveStatus = await githubApi.MigrationStatus(githubURL, githubSourceOrg, migrationId);
+
+                _log.LogInformation($"Waiting for archive generation to finish. Current status: {archiveStatus}");
+
+                if (archiveStatus == "exported")
+                {
+                    isFinished = true;
+                }
+                else if (archiveStatus == "failed")
+                {
+                    _log.LogError($"Archive generation failed with id: {migrationId}");
+                    return;
+                }
```

Comment:
We can return a strongly typed `enum` instead of strings from `MigrationStatus`

Reply:
Updated!